guides/release/routing/specifying-a-routes-model.md

```diff
@@ -163,11 +175,118 @@ each record in the song model and album model:
+});
+```
+
+Whatever shows up in the URL at the `:id`, the dynamic segment, will be available in the params for the route's `model` hook.
+
+```javascript {data-filename=app/routes/photo.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model(params) {
+    console.log('This came from the URL: ' + params.id)
+  }
+});
+```
+
+The `:id` name here isn't something special.
+Whatever name you give the dynamic segment, that's the key it will show up as in the `model` parameters.
+You could call it `:puppies` in `router.js` if you wanted to, and access it in the model as `params.puppies`.
+
+In the `model` hook for routes with dynamic segments, it's your job to
+use the ID (something like `47` or `post-slug`) when you make a data request.
+
+For example, if your app uses Ember Data to make API requests, you could include the ID in a `findRecord`:
+
+```javascript {data-filename=app/routes/photo.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model(params) {
+    this.store.findRecord('photo', params.id)
+  }
+});
+```
+
+### Linking to a dynamic segment
+
+There are two ways to link to a dynamic segment from an `.hbs` template using the
+[link-to](../../templates/links/)
+helper.
+Depending on which approach you use, it will affect whether that route's model hook is run.
+To learn how to link to a dynamic segment from within the JavaScript file, see the API documentation on
+[`transitionTo`](https://api.emberjs.com/ember/3.8/classes/RouterService/methods/transitionTo?anchor=transitionTo)
+instead.
+
+When you provide a string or number to the `link-to`, the dynamic segment's model hook will run when the app transitions to the new route.
+In this example, `photo.id` might have a value that's an ID like `4`:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo.id}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+However, if you provide the entire model context, the model hook for that URL segment will _not_ be run. In this example, we are passing the entire `photo` record:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+If a route you are trying to link to has multiple dynamic segments, be sure to specify all the necessary information for each segment. For example, a route like `/photos/4/comments/18` would need:
+
+```handlebars
+{{#link-to "photos.photo.comments.comment" 4 18}}
+  link text to display
+{{/link-to}}
+```
+
+Routes without dynamic segments will always execute the model hook.
+
 ## Reusing Route Context
 
 Sometimes you need to fetch a model, but your route doesn't have the parameters, because it's
 a child route and the route directly above or a few levels above has the parameters that your route
 needs.
+You might run into this if you have a URL like `/photos/4/comments/18`, and when you're in the comments route, you need a photo ID.
```

Comment:
when you're in the comments route, you need a photo `:id`

In this example, you are using the convention of `mode_id` instead of just plain `:id` ( i.e. `album_id`), this might be confusing for readers as the previous examples above are using `:id`?

Since the code example uses `song` and `album`, change the URL example to match that instead (e.g. `/albums/4/asongs/18`)

----
The section below was not edited so I can't add comments inline:

Firstly, talking to @mike-north he said that trying to get query params from the parents' route is an advanced concept, and an uncommon thing to do.  If we want to include the most important 20% of content that people need 80% of the time, this section does not pass the 80/20 rule. This looks like an an edge case and have "code smell".

Specific comments on why this section is confusing:

- [ ] Line 305 - 306:
> This is guaranteed to work because the parent route is loaded. But if you tried to do `paramsFor` on a sibling route, you wouldn't have the results you expected.

Q1:  Why do we have a guarantee with the parent route? Simply stating a guarantee is in one place but not the other doesn't help me form a mental model.
Q2: Why is using this method with a sibling route undesirable?

- [ ] Line 308
> This is a great way to use the parent context to load something that you want.
Q: What does the generic term "something that you want" mean?
In the example of `song` loading something from `album`, are we referring to data on the `album` e.g. album name? 

- [ ] Line 309 - 311
> Using `paramsFor` will also give you the query params defined on that route's controller.
This method could also be used to look up the current route's parameters from an action
or another method on the route, and in that case we have a shortcut: `this.paramsFor(this.routeName)`.

😩I have no idea what this paragraph means. 

- [ ] Line 313
> In our case, the parent route had already loaded its songs, so we would be writing unnecessary fetching logic.

There has been no explanation on the parent route and how it loads data so far, so how did we arrive at this conclusion that `songs` are already loaded?

- [ ] Line 329
> In the case above, the parent route looked something like this
Q1: Again, I'm still confused about what this example is
Q2: Why do we need to use `RSVP.hash` in the parent route?


I think either this entire section needs to be eliminated, or it need to be reviewed and updated to a sensible order:
- Start explaining from the "child route" (e.g. `song`) and move upwards (e.g. `album)
- Start explaining from the parent route and move downwards to child route

Reply:
I agree on a lot of these points. I didn't touch this section because it was super messy :P I would like to break this out into a new Issue/PR